runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -179,7 +180,7 @@ private boolean isSystemDoFn() {
    * @param <OutputT> the type of the {@link DoFn} (main) output elements
    */
   private static class DoFnContext<InputT, OutputT> extends DoFn<InputT, OutputT>.Context
-      implements DoFn.ExtraContextFactory<InputT, OutputT> {
+      implements Arguments<InputT, OutputT> {
```

Comment:
Arguments reads as really generic. I might prefer either `DoFn.Arguments` or `DoFnArguments` to clarify?